python/lib/core/dmod/core/common/collections/constants.py

```diff
@@ -0,0 +1,245 @@
+"""A generic indicator for the type of key in a map"""
+
+ValueType = typing.TypeVar("ValueType")
+"""A generic indicator for the type of value in a map"""
+
+
+class CollectionEvent(str, enum.Enum):
+    """
+    Defines the names of the types of events a collection might encounter
+    """
+    SET = "SET"
+    """The name of the event that would be triggered when setting a value on a collection"""
+
+    GET = "GET"
+    """The name of the event that would be triggered when getting a value from a collection"""
+
+    UPDATE = "UPDATE"
+    """The name of the event that would be triggered when modifying the value in a collection"""
+
+    EXTEND = "EXTEND"
+    """The name of the event that would be triggered when another collection is added to a collection"""
+
+    DELETE = "DELETE"
+    """The name of the event that would be triggered when an item is removed from a collection"""
+
+    ADD = "ADD"
+    """The name of the event that would be triggered when an item is added to a collection"""
+
+    INSERT = "INSERT"
+    """
+    The name of the event that would be triggered when an item is inserted into a collection at a specific location
+    """
+
+    POP = "POP"
+    """
+    The name of the event that would be triggered when a value is popped from one end of the collection
+    """
+
+    REMOVE = "REMOVE"
+    """
+    The name of the event that would be triggered when a value is simply removed from a collection 
+    (though not necessarily deleted)
+    """
+
+    REVERSE = "REVERSE"
+    """
+    The name of the event that would be triggered when the order of a collection is reversed
+    """
+
+    SORT = "SORT"
+    """
+    The name of the event that would be triggered when the order of items within a collection is modified based on 
+    a sorting operation
+    """
+
+    CLEAR = "CLEAR"
+    """
+    The name of the operation that would be triggered when all items are removed from a collection at once
+    """
+
+    @classmethod
+    def get(cls, name: str) -> CollectionEvent:
+        """
+        Get the enumerated value based on a similar name
+
+        Args:
+            name: The name of the CollectionEvent to find
+
+        Returns:
+            The matching CollectionEvent
+        """
+        lowercase_name = name.lower()
+        for entry in cls:
+            if entry.lower() == lowercase_name:
+                return cls[entry]
+        raise KeyError(f"There are no collection events named '{name}'")
+
+    @classmethod
+    def contains(cls, name: str) -> bool:
+        """
+        Check to see if there are any entries in the enum that matches the given name
+
+        Args:
+            name: The name of the value to find
+
+        Returns:
+            True if the value exists and won't throw an exception when 'get' is called
+        """
+        lowercase_name = name.lower()
+        return any(entry.lower() == lowercase_name for entry in cls)
+
+
+class MapHandler(typing.Generic[KeyType, ValueType]):
```

Comment:
It's for organization - you can use 'MapHandler.SET` as a constant for a type hint. For instance, `MapHandler.SET` is the type hint for an event handler for when `CollectionEvent.SET` is triggered on an instance of `typing.MutableMapping`. In this context, `SetHandler` would read as some form of handler for a `Set`.

Reply:
I mean at that point why dont you just make a module called `map_handlers` and put them all in there? That feels like it would have the highest cohesion.